scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Split.scala

```diff
@@ -55,60 +55,92 @@ case class Split(
+
+  def withOptimalTokenOpt(
+      token: => Option[Token],
+      killOnFail: Boolean = false
+  ): Split =
+    withOptimalAt(token.map(OptimalToken(_, killOnFail)))
 
-  def withOptimalToken(token: Token, killOnFail: Boolean = false): Split = {
-    require(optimalAt.isEmpty)
-    copy(optimalAt = Some(OptimalToken(token, killOnFail)))
+  def withOptimalToken(token: => Token, killOnFail: Boolean = false): Split =
+    withOptimalAt(Some(OptimalToken(token, killOnFail)))
+
+  def withOptimalAt(optimalAt: => Option[OptimalToken]): Split = {
+    require(this.optimalAt.isEmpty)
+    if (isIgnored) this else copy(optimalAt = optimalAt)
   }
 
-  def withPolicy(newPolicy: Policy): Split = {
+  def withPolicy(newPolicy: => Policy): Split = {
     if (policy != NoPolicy)
       throw new UnsupportedOperationException("Can't have two policies yet.")
-    copy(policy = newPolicy)
+    if (isIgnored) this else copy(policy = newPolicy)
   }
 
-  def withPolicy(newPolicy: Option[Policy]): Split =
-    newPolicy.fold(this)(withPolicy)
+  def withPolicyOpt(newPolicy: => Option[Policy]): Split =
+    if (isIgnored) this else newPolicy.fold(this)(withPolicy(_))
 
   def orElsePolicy(newPolicy: Policy): Split =
-    if (newPolicy.isEmpty) this
+    if (isIgnored || newPolicy.isEmpty) this
     else if (policy.isEmpty) copy(policy = newPolicy)
     else copy(policy = policy.orElse(newPolicy))
 
   def andThenPolicy(newPolicy: Policy): Split =
-    if (newPolicy.isEmpty) this
+    if (isIgnored || newPolicy.isEmpty) this
     else if (policy.isEmpty) copy(policy = newPolicy)
     else copy(policy = policy.andThen(newPolicy))
 
   def withPenalty(penalty: Int): Split =
-    copy(cost = cost + penalty)
+    if (isIgnored) this else copy(cost = cost + penalty)
 
   def withIndent(length: Length, expire: Token, expiresOn: ExpiresOn): Split =
     length match {
       case Num(0) => this
-      case _ => copy(indents = Indent(length, expire, expiresOn) +: indents)
+      case _ =>
+        if (isIgnored) this
+        else copy(indents = Indent(length, expire, expiresOn) +: indents)
     }
 
-  def sameSplit(other: Split): Boolean =
-    this.modification == other.modification &&
-      this.line.value == other.line.value && this.cost == other.cost
-
   override def toString = {
-    val prefix = if (ignoreIf) "!" else ""
+    val prefix = tag match {
+      case SplitTag.Ignored => "!"
+      case SplitTag.Active => ""
+      case _ => s"[$tag]"
+    }
     s"""$prefix$modification:${line.value}(cost=$cost, indents=$indentation, $policy)"""
   }
 }
 
 object Split {
 
   def ignored(implicit line: sourcecode.Line) =
-    Split(NoSplit, 0, ignoreIf = true)
+    Split(NoSplit, 0, tag = SplitTag.Ignored)
+
+  def onlyTaggedFor(tag: SplitTag.Value)(splits: Seq[Split]): Seq[Split] =
+    splits.flatMap { s =>
+      if (s.isTaggedFor(tag)) Some(s.copy(tag = SplitTag.Active)) else None
+    }
+}
 
+object SplitTag extends Enumeration {
```

Comment:
My only issue is `scala.Enumeration` usage

```scala
sealed trait SplitTag

object SplitTag {
  case object Ignored extends SplitTag
  case object Active extends SplitTag
}
```
sealed hierarchies is a little bit verbose, but a simple and reliable syntax.

Reply:
will do, thank you.